packages/salesforce/src/Adaptor.js

```diff
@@ -148,16 +148,23 @@ export function retrieve(sObject, id, callback) {
  * Execute an SOQL query.
  * Note that in an event of a query error,
  * error logs will be printed but the operation will not throw the error.
+ *
+ * This function makes an API call to retrieve query results.
+ * It is subject to rate limits imposed by the Salesforce API.
+ * `Maximum rows returned in SOQL query results`: </br>
+ * This function will return `2,000` results per request (API v28.0+),
+ * `200` results (Previous API versions)
  * @public
  * @example
  * query(state=> `SELECT Id FROM Patient__c WHERE Health_ID__c = '${state.data.field1}'`);
  * @example <caption>Query more records if next records are available</caption>
  * query(state=> `SELECT Id FROM Patient__c WHERE Health_ID__c = '${state.data.field1}'`, { autoFetch: true });
  * @function
- * @param {String} qs - A query string.
+ * @param {String} qs - A query string. Must be less than `4000` characters in WHERE clause
```

Comment:
Just the where clause? Not the whole query?

I think this might be buried out of sight here - would a dev ever see it? Maybe there's not much more we can do, let me think on it